src/libcore/iter/mod.rs

```diff
@@ -645,6 +645,21 @@ impl<I> Iterator for Cycle<I> where I: Clone + Iterator {
     }
 }
 
+#[stable(feature = "rust1", since = "1.0.0")]
+impl<I> Iterator for Cycle<I>
+where I: Clone + Iterator + ExactSizeIterator,
+      <I as Iterator>::Item : Clone + Copy {
```

Comment:
This should be just `I::Item: Copy` style wise. `Iterator + ExactSizeIterator` is also redundant, just `ExactSizeIterator` is the same.